File: LinearLayout80/app/src/main/java/com/example/linearlayout80/MainActivity.java
package com.example.linearlayout80;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
//import android.widget.Button;
//import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    //Button btn = null;
    //EditText et_name = null;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        //activity_main.xml에서 그려진 버튼을 MainAcivity.java 에서 접근할 수 있다.
        //id를 선언하면 findViewById(R.id.xml에서 선언한 이름)
        //R.java는 안드로이드 스튜디오에서 자동으로 관리하는 클래스 로서
        //xml에서 그려진 화면 레이아웃 또는 버튼 같은 것들을 주소번지를 통해서 접근한다.
        //차이점은 아이디로 접근한다는 것이 다름
        //btn = findViewById(R.id.btn);
        //et_name = findViewById(R.id.et_name);
        //et_name.setText("테스트");
        //xml로 그려진 화면을 호출하는 것은 steContentView메소드의 파라미터로
        //xml문서의 이름을 사용하여 접근 호출함.
        setContentView(R.layout.activity_main);
    }
    public void send(View v){
        Toast toast = Toast.makeText(MainActivity.this,
                "전송호출",Toast.LENGTH_SHORT);
        toast.show();
    }
}

File: Lotto80/app/src/main/java/com/example/lotto80/MainActivity.kt
package com.example.lotto80

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.NumberPicker

class MainActivity : AppCompatActivity() {
    //선언 뒤에 lazy가 오면 activity_main4.xml을 스캔할 때 까지 기다린다.
    //$(document).ready(function(){
    //      $("dg_dept").datagrid({
    //              url:"XXX.jsp"
    //      });
    //});
    private val runButton: Button by lazy {
        findViewById(R.id.runButton)
    }
    //NumberPicker의 범위를 정해주자. 지금은 0만 보임. 움직이지도 않음
    private val numberPicker: NumberPicker by lazy{
        findViewById(R.id.numberPicker)
    }
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        numberPicker.minValue = 1
        numberPicker.maxValue = 45

        //자동채번 버튼 구현
        initRunButton()
    }

    private fun initRunButton(){
        runButton.setOnClickListener{
            val list = getRandomNumber()
            Log.d("MainActivity", list.toString())
        }
    }

    override fun onStart(){
        super.onStart()
        //실행이 되어있다가 (백스텍에 머물러 있다가) 다시 소환될때 onCreate를 타는게 아니라 여기로 온다.
    }
    //채번 알고리즘
    private fun getRandomNumber():List<Int>{
        //자바의 apply => Integer name[] = new Integer[]{1,2,3,4,5}
        //자바라면 => List<Integer> num = new ArrayList<>();
        val numberList = mutableListOf<Int>()
            .apply {
                for (i in 1..45) {
                    this.add(i)
                }
            }
        numberList.shuffle() //1~45 숫자가 섞임
        val newList = numberList.subList(0,6) //배열0~6까지 자른단 얘기임
        return newList.sorted() //정렬시켜서 내보냄
    }
}